### DISCUSSION TITLE
Using docker secrets or environment variables in the configuration

### DISCUSSION THREAD

**Discussion by retrodaredevil** [gethomepage/homepage#1782] [Support] [answered]
Hello. I recently setup Homepage and am going through the documentation.

I was unable to find documentation on how to include a docker secret in my yaml configuration or how to include environment variables. In the case of a docker secret, it would be ideal to be able to include the output of a file such as `/run/secrets/file`. I wasn't sure if that was a supported feature or not, which is why I'm also wondering what the syntax for including environment variables is.

**Comment by retrodaredevil** [comment, marked as answer]
I ended up looking at the source code, since I couldn't find anything about environment variable substitution.
To get this working, you may place text like `{{HOMEPAGE_VAR_XXX}}`, to directly substitute environment variables. If you define an environment variable named `{{HOMEPAGE_FILE_XXX}}`, the contents of the file at the location specified by the environment variable will be substituted.

I think this should be better documented in the docs.

Here is the code: https://github.com/benphelps/homepage/blob/63d4c6bdf8474299316965f655fc092656f50154/src/utils/config/config.js#L51C22-L51C22

**Reply by cesarnr21** [reply]
I actually have been trying to get this to work in my setup, but it isn't working for some reason.

- I set up `HOMEPAGE_VAR_WEATHER_API="whatever"` in a `.env` file.
- I defined the path in the `compose.yaml` file like this 
  ```
  env_file:
        - path: path/.env
  ```

- and in the `widgets.yaml` set the api key to `apiKey: {{HOMEPAGE_VAR_WEATHER_API}}`

I keep getting an API Error anytime I reload homepage

**Reply by shamoon** [reply]
Open a new discussion and provide detailed info. The bot will probably be able to help.
